### PR TITLE
Add Configurations and frontend changes

### PR DESCRIPTION
**Pull request by vijay-webkul**
## Issue Reference
#7410
-> Add new feature in configuration and show on the frontend page

### CODE REVIEW

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Comment by Sanjaybhattwebkul** [comment]
@vijay-webkul 
This is **working fine.**

**Comment by Amitk-webkul** [comment]
@vijay-webkul 
now It is working fine. [Here](https://prnt.sc/RxFYSW2Ryhsp)

File: packages/Webkul/Admin/src/Config/system.php [hunk 1]
```diff
@@ -132,14 +132,35 @@
...
+                'title'      => 'admin::app.admin.system.allow-no-of-related-productpage',
+                'type'       => 'number',
+                'validation' => 'min:0',
+            ], [
+                'name'       => 'no_of_up_sells_product_productpage',
+                'title'      => 'admin::app.admin.system.allow-no-of-up-sells-productpage',
+                'type'       => 'number', 
+                'validation' => 'min:0',
+            ], [
+                'name'       => 'no-of-cross-sells-product_productpage',
```
Comment (vivek-webkul) [inline packages/Webkul/Admin/src/Config/system.php:158]: Use Snake casing.

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/cross-sells.blade.php [hunk 2]
```diff
@@ -1,15 +1,9 @@
-@foreach ($cart->items as $item)
-    @php
-        $product = $item->product;
+@php
+$productCount = core()->getConfigData('catalog.products.product-view-page.no-of-cross-sells-product_productpage');
```
Comment (vivek-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view/cross-sells.blade.php:2]: correct the alignment and use snake casing in config field.

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/related-products.blade.php [hunk 3]
```diff
@@ -1,4 +1,7 @@
-@php $relatedProducts = $product->related_products()->get(); @endphp
+@php
+$productCount = core()->getConfigData('catalog.products.product-view-page.no_of_related_product_productpage');
```
Comment (vivek-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view/related-products.blade.php:2]: correct the alignment.

**Review by vivek-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Admin/src/Config/system.php (hunk 1 above)
Comment (vijay-webkul) [reply packages/Webkul/Admin/src/Config/system.php:158]: resolved

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/cross-sells.blade.php (hunk 2 above)
Comment (vijay-webkul) [reply packages/Webkul/Velocity/src/Resources/views/shop/products/view/cross-sells.blade.php:2]: resolved

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/related-products.blade.php (hunk 3 above)
Comment (vijay-webkul) [reply packages/Webkul/Velocity/src/Resources/views/shop/products/view/related-products.blade.php:2]: resolved

**Review by Sanjaybhattwebkul — APPROVED** [review]

**Review by vivek-webkul — APPROVED** [review]

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/cross-sells.blade.php [hunk 4]
```diff
@@ -1,15 +1,9 @@
-@foreach ($cart->items as $item)
-    @php
-        $product = $item->product;
+@php
+    $productCount = core()->getConfigData('catalog.products.product-view-page.no_of_cross_sells_product_productpage'); 
+    $products     = $product->cross_sells()->take($productCount)->get();
+@endphp
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view/cross-sells.blade.php:4]: Move all these logic to controller.

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/related-products.blade.php [hunk 5]
```diff
@@ -1,4 +1,7 @@
-@php $relatedProducts = $product->related_products()->get(); @endphp
+@php
+    $productCount    = core()->getConfigData('catalog.products.product-view-page.no_of_related_product_productpage'); 
+    $relatedProducts = $product->related_products()->take($productCount)->get();
+@endphp
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view/related-products.blade.php:4]: Move all these logic to controller.

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view/up-sells.blade.php [hunk 6]
```diff
@@ -1,6 +1,9 @@
-@php $productUpSells = $product->up_sells()->get(); @endphp
+@php 
+    $productCount   = core()->getConfigData('catalog.products.product-view-page.no_of_up_sells_product_productpage'); 
+    $productUpSells = $product->up_sells()->take($productCount)->get();
+@endphp
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view/up-sells.blade.php:4]: Move all these logic to controller.

File: packages/Webkul/Admin/src/Resources/lang/en/app.php [hunk 7]
```diff
@@ -1677,6 +1679,9 @@
             'show-search-input-field'               => 'Show Search Input Field',
             'allow-no-of-new-product-homepage'      => 'Allowed number of New Product in Homepage',
             'allow-no-of-featured-product-homepage' => 'Allowed number of Featured Product in Homepage',
+            'allow-no-of-related-productpage'       => 'Allowed number of Related Product in Productpage',
+            'allow-no-of-up-sells-productpage'      => 'Allowed number of Up-Sells Product in Productpage',
+            'allow-no-of-cross-sells-productpage'   => 'Allowed number of Cross-Sells Product in Productpage',
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/src/Resources/lang/en/app.php:1684]: Earlier we are using the `kebab-case` but on improvement, we are focussing full array keys as `snake_case`. So add these as `snake_case` and the other one we will improvise in a single PR.

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view.blade.php [hunk 8]
```diff
@@ -197,8 +197,8 @@
 
         <div class="related-products">
             @include('shop::products.view.related-products')
-            
             @include('shop::products.view.up-sells')
+            @include('shop::products.view.cross-sells')
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view.blade.php:201]: Add one line here.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
Check the comments.

**Comment by vijay-webkul** [comment]
working on this

**Comment by Sanjaybhattwebkul** [comment]
@vijay-webkul 
Any update ?

**Pull request closed without merging**